pkg/apis/serving/v1alpha1/service_defaults_test.go

```diff
@@ -264,6 +275,190 @@ func TestServiceDefaulting(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "inline defaults to run latest",
+		in: &Service{
+			Spec: ServiceSpec{
+				ConfigurationSpec: ConfigurationSpec{
+					RevisionTemplate: &RevisionTemplateSpec{
+						Spec: RevisionSpec{
+							RevisionSpec: v1beta1.RevisionSpec{
+								PodSpec: v1beta1.PodSpec{
+									Containers: []corev1.Container{{
+										Image: "blah",
+									}},
+								},
+							},
+						},
+					},
+				},
+				// No RouteSpec should get defaulted to "run latest"
+			},
+		},
+		want: &Service{
+			Spec: ServiceSpec{
+				ConfigurationSpec: ConfigurationSpec{
+					RevisionTemplate: &RevisionTemplateSpec{
+						Spec: RevisionSpec{
+							RevisionSpec: v1beta1.RevisionSpec{
+								TimeoutSeconds: ptr.Int64(config.DefaultRevisionTimeoutSeconds),
+								PodSpec: v1beta1.PodSpec{
+									Containers: []corev1.Container{{
+										Image:     "blah",
+										Resources: defaultResources,
+									}},
+								},
+							},
+						},
+					},
+				},
+				RouteSpec: RouteSpec{
+					Traffic: []TrafficTarget{{
+						TrafficTarget: v1beta1.TrafficTarget{
+							LatestRevision: ptr.Bool(true),
+							Percent:        100,
+						},
+					}},
+				},
+			},
+		},
+	}, {
+		name: "inline with empty RevisionSpec",
+		in: &Service{
+			Spec: ServiceSpec{
+				ConfigurationSpec: ConfigurationSpec{
+					RevisionTemplate: &RevisionTemplateSpec{},
+				},
+				// No RouteSpec should get defaulted to "run latest"
+			},
+		},
+		want: &Service{
+			Spec: ServiceSpec{
+				ConfigurationSpec: ConfigurationSpec{
+					RevisionTemplate: &RevisionTemplateSpec{
+						Spec: RevisionSpec{
+							RevisionSpec: v1beta1.RevisionSpec{
+								TimeoutSeconds: ptr.Int64(config.DefaultRevisionTimeoutSeconds),
+							},
+							Container: &corev1.Container{
+								Resources: defaultResources,
+							},
+						},
+					},
+				},
+				RouteSpec: RouteSpec{
+					Traffic: []TrafficTarget{{
+						TrafficTarget: v1beta1.TrafficTarget{
+							LatestRevision: ptr.Bool(true),
+							Percent:        100,
+						},
+					}},
+				},
+			},
+		},
+	}, {
+		name: "inline defaults to run latest (non-nil traffic)",
+		in: &Service{
```

Comment:
can we use constructors? This depth is mind boggling :)

Reply:
I think it's useful to be explicit in at least some of our table tests.  I often wish it was easier to see the resolved inputs and outputs of our Reconciler table tests on failures (maybe we should dump the fully resolved outputs on failure?), but these are still a far cry from where those were when we went with constructors (several screens per "row").  🤷‍♂️

Reply:
🍂 
okay